### PR TITLE
XML signing with JKS

### PR DESCRIPTION
**Pull request by viacheslav-rud**
## Description of changes in release / Impact of release:
(insert text here)

## Documentation
(insert text here)

### Is this a breaking change?
- [ ] Yes
- [ ] No

### If you answered Yes then describe why is it so
(insert text here if applicable)

### Is there a way to disable the change?
- [ ] Use previous release
- [ ] Use a feature flag
- [ ] No

#### Additional details go here
(insert text here if applicable)

### CODE REVIEW

**Comment by viacheslav-rud** [comment]
I need to cover few changes with tests but the general implementation is ready

**Comment by viacheslav-rud** [comment]
Indentation doesn't look to be working propely so CI will fail

File: larky/src/main/java/com/verygood/security/larky/modules/jks/JKS.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                certificateChain.add(larkyCertificate);
+            }
+        } catch (KeyStoreException | NoSuchAlgorithmException | UnrecoverableKeyException | IOException |
+                 CertificateException e) {
+            throw new EvalException(e);
+        }
+
+        return Tuple.of(
+                keyPair,
+                certificateChain.remove(0),
```
Comment (nestorsokil) [inline larky/src/main/java/com/verygood/security/larky/modules/jks/JKS.java:90]: a bit unclear what's happening here

Comment (viacheslav-rud) [reply larky/src/main/java/com/verygood/security/larky/modules/jks/JKS.java:90]: Basically we emulate behaviour from Python library, I think putting a link to the Python code should be good

Comment (viacheslav-rud) [reply larky/src/main/java/com/verygood/security/larky/modules/jks/JKS.java:90]: This is actually an analogy of this method but with JKS under the hood
https://github.com/verygoodsecurity/starlarky/blob/master/larky/src/main/java/com/verygood/security/larky/modules/openssl/OpenSSL.java#L242-L257

File: larky/src/main/resources/stdlib/xml/etree/ElementTree.star [hunk 2]
```diff
@@ -2151,14 +2177,18 @@ def XMLParser(html=0, target=None, encoding=None):
 
     def feed(data):
         """Feed encoded data to parser."""
-        rval = Result.Ok(self.parser.Parse).map(lambda f: f(data, 0))
+        # rval = Result.Ok(self.parser.Parse).map(lambda f: f(data, 0))
```
Comment (Iapetus999) [inline larky/src/main/resources/stdlib/xml/etree/ElementTree.star:2180]: rm comment

Comment (viacheslav-rud) [reply larky/src/main/resources/stdlib/xml/etree/ElementTree.star]: Done for all the comments

File: larky/src/main/resources/vgs/util/crypt.star [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+load("@stdlib//base64", base64="base64")
+load("@stdlib//io", io="io")
+load("@vgs//jks/jks2pycryptodome", jks="jks2pycryptodome")
+
+load("@vendor//cryptography/hazmat/backends/pycryptodome", backend="backend")
+load("@stdlib//xml/etree/ElementTree", etree="ElementTree")
+load("@vgs//xmlsig-java-compatible", xmlsig="xmlsig")
+
+
+def mcInControlSignature(xml_string, sign_element_xpath, keystore_base64, keystore_password, key_alias, key_password):
```
Comment (Agitolyev) [inline larky/src/main/resources/vgs/util/crypt.star:11]: Please, add doc.

File: larky/src/main/resources/vgs/xmlsig-java-compatible/__init__.star [hunk 4]
```diff
@@ -0,0 +1,17 @@
+load("@stdlib//larky", larky="larky")
+
+load("@vgs//xmlsig-java-compatible/algorithms", algorithms="algorithms")
+load("@vgs//xmlsig-java-compatible/constants", constants="constants")
+load("@vgs//xmlsig-java-compatible/ns", ns="ns")
+load("@vgs//xmlsig-java-compatible/template", template="template")
+load("@vgs//xmlsig-java-compatible/signature_context", SignatureContext="SignatureContext")
+
+
+xmlsig = larky.struct(
```
Comment (Agitolyev) [inline larky/src/main/resources/vgs/xmlsig-java-compatible/__init__.star:10]: Please, port test for original lib and adapt for java scenario.

File: larky/src/test/resources/vgs_tests/xmlsig/base.star [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    xml = parse_xml(name)
+
+    # Stringify the root, <Envelope/> nodes of the two documents.
+    expected_text = etree.tostring(xml, pretty_print=False)
+    result_text = etree.tostring(result, pretty_print=False)
+    # Compare the results.
+    if expected_text != result_text:
+        print("expected: ", expected_text, sep="\n")
+        print("result: ", result_text, sep="\n")
+    asserts.eq(expected_text, result_text)
```
Comment (Agitolyev) [inline larky/src/test/resources/vgs_tests/xmlsig/base.star:29]: Please, add newline.

File: larky/src/test/resources/vgs_tests/xmlsig/test_default_signer.star [hunk 6]
```diff
@@ -0,0 +1,24 @@
+load("@stdlib//unittest", unittest="unittest")
```
Comment (Agitolyev) [inline larky/src/test/resources/vgs_tests/xmlsig/test_default_signer.star:1]: Is it in the right dir structure?

File: larky/src/main/java/com/verygood/security/larky/modules/vgs/jks/JKSModule.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+@StarlarkBuiltin(
+        name = "jks",
+        category = "BUILTIN",
+        doc = "")
+public class JKSModule implements StarlarkValue {
+
+    public static final JKSModule INSTANCE =  new JKSModule();
+
+    @StarlarkMethod(name = "JKS", doc = "Java KeyStore module", structField = true)
```
[the remark this answers is not in the document]
Comment (viacheslav-rud) [reply larky/src/main/java/com/verygood/security/larky/modules/vgs/jks/JKSModule.java]: Added

File: larky/src/test/resources/stdlib_tests/xml/etree/test_elementtree.star [hunk 8]
```diff
@@ -609,6 +609,11 @@ def test_namespace_prefix():
     ET._namespace_map.clear()
     ET._namespace_map.update(nsmap)
 
+def test_parse_and_getroot():
```
Comment (Agitolyev) [inline larky/src/test/resources/stdlib_tests/xml/etree/test_elementtree.star:612]: Will it be reasonable to have one test per method?

Comment (viacheslav-rud) [reply larky/src/test/resources/stdlib_tests/xml/etree/test_elementtree.star:612]: Not really, I actually took it from the original CPython code

**Pull request merged**